### DISCUSSION TITLE
User-defined template folder (config property)

### DISCUSSION THREAD

**Discussion by kg-intec** [JakubAndrysek/MkDoxy#48] [Ideas]
Sorry its me again 😅

After presenting my generated doxygen to my colleagues it was decided to use MkDoxy to autogenerate our documentation. And then the bikesheding started, basically everybody had some small wishes / comments on how to change the general output. So I looked into your code and saw, that I could basically adjust the templates myself (as long as I have access to the template directory in the python environment). But I cannot modify the environment on other workstations or our buildserver (due to missing rights... and also because the admin wanted to hit me with a stick after I told him what I was going to do..). 

So it would be nice to have a config parameter to change the template directory to a user defined one, or (advanced) to overwrite only specific templates.

**Comment by JakubAndrysek** [comment]
Thank you for the next feedback. 
Right now I didn`t have so much time to work on this project. 
I'll get to this project in May at the earliest, so if you have an idea of how to add it, your PRs will be welcome.

Have a nice day 
Jacob

**Comment by kg-intec** [comment]
I am an ok python programmer (my work is in C++ / C#). So I will try to help were I can... For the start I will try to implement this config parameter 😀. The good thing is that a colleague is a Python programmer, whom I will annoy with questions if I have any questions, and that I can allocate a small part of my working time to this because it is helpful for the project...  

So maybe I can be useful 😎,

Kai

**Comment by JakubAndrysek** [comment]
Can I ask what project you are using it for? Is public or private and in which country are you developing?

Thanks

**Comment by kg-intec** [comment]
I work for a midsize company in germany (about 100 employees). We develop software for virtual comissioning simple to complex production plants. Which basically means real time simulation and data analysis for robots and conveyor belts and stuff... We are modernizing our internal processes and the developer documentation will be switched from a 20-year-old MediaWiki to internal documentation with Mkdocs / Doxygen etc and a knowledgebase, whereby I will automatically feed the knowledgebase with the articles from the docs.

**Comment by kg-intec** [comment]
I prepared a simple PR to add the `template-dir` parameter to the project configuration to allow the user to customize each of their projects according to their needs. It includes the code changes and a mention in the README. I did not add another or modify an example.

If you want me to adapt an example or you need other things for a PR, just write 😀. You can find the current code [here](https://github.com/kg-intec/MkDoxy/tree/userdefined-templates).

**Comment by JakubAndrysek** [comment]
#39 Closed with this PR

**Comment by JakubAndrysek** [comment]
Thank you [kg-intec](https://github.com/kg-intec) for the help with the PR.
I have changed the behaviour and custom template files will overwrite the default templates.
So you don't need to copy the whole default template file to your custom template folder.

Is it ok for you? @kg-intec
